src/gtc/passes/oir_optimizations/caches.py

```diff
@@ -105,7 +105,9 @@ class KCacheDetection(NodeTranslator):
     max_cacheable_offset: int = 5
 
     def visit_VerticalLoop(self, node: oir.VerticalLoop, **kwargs: Any) -> oir.VerticalLoop:
-        if node.loop_order == common.LoopOrder.PARALLEL:
+        if node.loop_order == common.LoopOrder.PARALLEL or any(
```

Comment:
Please add a comment why limiting to a single horizontal_execution is a requirement. Is it because of the kcache living in the scope of the horizontal execution?

Reply:
Yes, exactly. Usually this restriction should not matter at all, as all horizontal executions without horizontal offsets should be merged. Currently though this is not possible as soon as there are masks, because there are no if statements yet.

Reply:
Please add this info as a comment.

Reply:
Done.